tscreen.go

```diff
@@ -1804,6 +1847,7 @@ func (t *tScreen) engage() error {
 	t.stopQ = stopQ
 	t.enableMouse(t.mouseFlags)
 	t.enablePasting(t.pasteEnabled)
+	t.enableFocusReporting()
```

Comment:
Is it ok to do this unconditionally, or do we need a `t.focusReportingEnabled` flag to opt in?

Reply:
I have mixed feelings.  I think it's probably ok to be unconditional -- hopefully applications will know to ignore events that they don't understand.

Reply:
Sorry to jump in out of the blue, but if this code enables focus reporting unconditionally for all applications, I think it'd be better to make it opt-in. Focus reporting has side-effects if, for example, if you are using a program that enabled focus reporting over ssh and the ssh connection drops. The user is then surprised to have text printed on their screen (probably in their shell) any time they focus or unfocus the window.

This could be considered a bug in ssh, or in the shells that don't know how to interpret the sequence, but it's pervasive. I think it's present in mosh. In any case, tmux does not turning focus reporting on unless explicitly asked to, and I think most applications follow this approach. Having focus reporting be turned off by default should be possible for applications using `tcell`, IMO.

Apologies if I misunderstood what this code does.

Reply:
I understand your concern, but I don't think making focus reporting opt in helps in any way.

I simulated your ssh scenario by simply `kill -9` a tcell application (lazygit in my case, but you could also use `_demos/mouse.go`). If you `kill -9` it, the terminal is in a very messed up state anyway, it spews mouse reporting escape codes and other things, and users just have to know that they need to type `reset` in such a case to get back to a normal terminal. Whether we also keep sending focus events or not doesn't really make a big difference in this case.

Reply:
OK, I take that back. 😄  I didn't realize that mouse reporting is also opt-in, so only applications that use it suffer from the problem. It totally makes sense then to treat focus reporting in the same way. See 89aca70d823.

Reply:
Thanks, I appreciate it! I was going to pin down a way to demonstrate where this makes a difference and make a screencast. I don't have to now, but I know it does make a difference with the way I use ssh/mosh. I believe making this opt-in will help many programs and users that don't know about focus reporting.

Also, thanks for implementing this feature in the first place! 🎉